api/v1/address.go

```diff
@@ -1,55 +1,104 @@
 package api
 
 import (
+	"encoding/json"
 	"net/http"
 
+	"github.com/data-preservation-programs/spade-tenant/db"
 	"github.com/labstack/echo/v4"
+	"gorm.io/gorm"
 )
 
-type Addresses []Address
-type AddressesMutable []AddressMutable
+type Addresses []db.Address
+type Address db.Address
 
-type Address struct {
-	AddressMutable
-	ActorID *uint `json:"actor_id,omitempty"` // TODO :swagger docs should be null not 0
+type AddressResponse struct {
+	AddressRobust    string `json:"address_robust" gorm:"primaryKey"`
+	AddressActorID   uint   `json:"actor_id"`
+	AddressIsSigning bool   `json:"is_signing" gorm:"default:true;not null"`
 }
 
-type AddressMutable struct {
-	Address   string `json:"address"`
-	IsSigning bool   `json:"is_signing"` // true - active dealmaking from this address, false - still counts as an associated wallet
+func ConfigureAddressesRouter(e *echo.Group, service *db.SpdTenantSvc) {
+	g := e.Group("/addresses")
+	g.PUT("", handleSetAddresses)
+	g.DELETE("", handleDeleteAddresses)
+	g.GET("", handleGetAddresses)
 }
 
 // handleSetAddresses godoc
 //
-//			@Summary		Update addresses associated with a tenant
-//	 		@Security apiKey
-//		  @Param 			addresses body AddressesMutable true "New addresses to add or change is_signing flag of"
-//			@Produce		json
-//			@Success		200	{object}	ResponseEnvelope{response=Addresses}
-//			@Router			/addresses [put]
+//	@Summary		Update addresses associated with a tenant
+//	@Param			token header string true "Auth token"
```

Comment:
probably a mistake changing to `@Param token header`. should keep it as `@Security apiKey` as that's needed for proper docs site generation

Reply:
yeah looks like my stacked diff trashed it for some reason. Ill update all the prs on the stack